Update docs on migrating NFS home directories (#753)
2i2c-org/infrastructure: docs/howto/operate/move-hub.md

Primarily used with GKE right now.

> 1. SSH into **both** the target server and source server and then enable the `ubuntu` user on **both** servers with `sudo su ubuntu`.

Reviewer: Might be a stupid question, but do our clusters all have an `ubuntu` user on them?

Author: It's not our clusters but a VM that is either acting as an NFS server or a VM that has some kind of storage mounted (GCP Filestore, Azure Filestore, etc).

I _think_ most linux distros ship with an `ubuntu` user by default - I didn't have to `sudo useradd ubuntu` - but I'm not 100% sure.

   This will ensure that the keys we are about to create will be assigned to the correct user.

> 2. In the target NFS server, create a new ssh key-pair, with

Reviewer 2: Don't you need to set up some specific permissions after creating the key?

Author: No, by making sure you activate the `ubuntu` user first and using the `ssh-keygen` command, all the permissions are taken care of.

   `ssh-keygen -f nfs-transfer-key`

> 3. Append the public key `nfs-transfer-key.pub` to the source NFS
>    server's `/home/ubuntu/.ssh/authorized_keys` file, so the target
>    NFS server can open SSH connections to the source NFS server.

Reviewer: What do you think about splitting the sentence? (I read it 2 times until I made a visual model and was sure I understood :sweat_smile: )
```suggestion
3. Append the public key `nfs-transfer-key.pub` to the source NFS
   server's `/home/ubuntu/.ssh/authorized_keys` file. This way, the target
   NFS server will be able to open SSH connections to the source NFS server.
```

4. Copy the NFS home directories from the source NFS server to
   the target NFS server, making sure that the NFS exports locations
   match up appopriately. For example, if the source NFS server has
   home directories for each hub stored in `/export/home-01/homes`,
   and the target NFS server also has hub home directories stored under
   `/export/home-01/homes`, you can `scp` the contents across with:

   ```bash
   scp -p -r -i nfs-transfer-key ubuntu@nfs-source-server-public-IP:/export/home-01/homes/<hub-name> /export/home-01/homes/<hub-name>
   ```

   This makes sure the target is owned by the `ubuntu` user, which has
   uid `1000`. Our user pods run as uid `1000`, so this makes sure they
   can mount their home directories.

As an alternative to `scp` you can use `rsync` as follows:

```bash
rsync -e 'ssh -i nfs-transfer-key' -rougvhP ubuntu@nfs-source-server-public-IP:/export/home-01/homes/<hub-name>/ /export/home-01/homes/<hub-name>/

> ```

Reviewer 2: Could it be the case the `-e 'ssh -i nfs-transfer-key'` is not actually needed if you ssh-add the key?

Author: I don't know, I'm not familiar enough with `ssh add`. What I found is that because I prefaced the command with `sudo`, `rsync` was looking at `root`'s ssh keys, so I was just specific in saying which key I wanted to use.
